pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +290,66 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeatLoop() {
+	heartbeatInterval := pm.conf.HeartBeatInterval
+
+	if pm.registrationInfo.CollectorCredentialId == "" || pm.registrationInfo.CollectorId == "" {
+		pm.logger.Error("Collector not registered, cannot send heartbeat")
+		return
+	}
+
+	pm.logger.Info("Heartbeat heartbeat API initialized. Starting sending hearbeat requests")
+	for {
+		select {
+		case <-pm.closeChan:
+			pm.logger.Info("Heartbeat sender turn off")
+			return
+		default:
+			err := pm.sendHeartbeat()
+			if err != nil {
+				pm.logger.Error("Heartbeat error: ", zap.String("error: ", err.Error()))
+			}
+			pm.logger.Debug("Heartbeat sent")
+			//nolint // pmatyjasek: Bypass rule S1037
+			select {
+			case <-time.After(heartbeatInterval):
+			}
+		}
+	}
+}
+
+func (pm *sumologicExtension) sendHeartbeat() error {
+	u, err := url.Parse(pm.baseUrl + heartbeatUrl)
+	if err != nil {
+		return fmt.Errorf("unable to parse heartbeat URL %s", err.Error())
+	}
+	req, err := http.NewRequest(http.MethodPost, u.String(), nil)
+	if err != nil {
+		return fmt.Errorf("unable to create HTTP request %s", err.Error())
+	}
+
+	pm.addCollectorCredentials(req)
+	pm.addJSONHeaders(req)
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("unable to send HTTP request: %w", err)
+	}
+	defer res.Body.Close()
+	if res.StatusCode != 204 {
+		var buff bytes.Buffer
+		if _, err := io.Copy(&buff, res.Body); err != nil {
+			return fmt.Errorf(
+				"failed to copy collector heartbeat response body, status code: %d, err: %w",
+				res.StatusCode, err)
```

Comment:
```suggestion
			return fmt.Errorf(
				"failed to copy collector heartbeat response body, status code: %d, err: %w",
				res.StatusCode, err,
			)
```

oh no the style debate once again 🤯